modules/sql/src/main/scala/gem/sql/enum/GsaoiEnums.scala

```diff
@@ -0,0 +1,45 @@
+// Copyright (c) 2016-2019 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gem.sql
+package enum
+
+import doobie._, doobie.implicits._
+import java.time.Duration
+import shapeless.record._
+import shapeless.Witness
+
+@SuppressWarnings(Array("org.wartremover.warts.NonUnitStatements"))
+object GsaoiEnums {
+  import EnumRefs._
+
+  val enums: List[ConnectionIO[EnumDef]] =
+    List(
+
+      EnumDef.fromQuery("GsaoiReadMode", "GSAOI Read Mode") {
+        type E = Record.`'tag -> String, 'shortName -> String, 'longName -> String, 'ndr -> Int, 'readNoise -> Int, 'minimumExposureTime -> Duration, 'overhead -> Int`.T
+        sql"""SELECT id, id tag, short_name, long_name, ndr, read_noise, minimum_exposure_time, overhead FROM e_gsaoi_read_mode""".query[(String, E)]
+      },
+
+      EnumDef.fromQuery("GsaoiFilter", "GSAOI Filter") {
+        val  m = Witness('MagnitudeBand)
+        val  r = Witness('GsaoiReadMode)
+        type M = m.T
+        type R = r.T
+        type E = Record.`'tag -> String, 'shortName -> String, 'longName -> String, 'wavelength -> Double, 'readMode -> EnumRef[R], 'expsoureTime5050 -> Double, 'exposureTimeHalfWell -> Double, 'band -> Option[EnumRef[M]]`.T
```

Comment:
This is where you'd map the wavelength to a `Wavelength.Pm`, `Wavelength.Nm`, or `Wavelength.Um` depending on the units.